### PR TITLE
Add api support for non-collections

### PR DESCRIPTION
**Pull request by eatonphil**
Runner-up to the account dropdown PR

### CODE REVIEW

File: test/linodes/linode/settings/layouts/DisplayPage.spec.js [hunk 1]
```diff
@@ -71,9 +71,9 @@ describe('linodes/linode/settings/layouts/DisplayPage', () => {
 
     const label = page.find('.form-group').at(1);
     expect(label.find('.form-control-feedback')
-      .children()
-      .first()
-      .text()).to.equal(error);
+                .children()
+                .first()
+                .text()).to.equal(error);
```
Comment (na3d) [inline test/linodes/linode/settings/layouts/DisplayPage.spec.js]: Are these extra spaces the linter is allowing?

Comment (eatonphil) [inline test/linodes/linode/settings/layouts/DisplayPage.spec.js]: Apparently

File: src/router.js [hunk 2]
```diff
@@ -67,14 +69,19 @@ export class LoadingRouterContext extends RouterContext {
     const { location: { pathname }, params } = this.props;
     // No need to fetch these in weblish.
     if (!(pathname.endsWith('/weblish') && params.linodeLabel)) {
-      store.dispatch(kernels.all());
-      store.dispatch(types.all());
-      store.dispatch(regions.all());
-      store.dispatch(distributions.all());
+      await Promise.all([
```
Comment (mparke) [inline src/router.js]: Promise.all woooo

File: src/router.js [hunk 3]
```diff
@@ -67,14 +69,19 @@ export class LoadingRouterContext extends RouterContext {
...
+        store.dispatch(account.one()),
+        store.dispatch(kernels.all()),
+        store.dispatch(types.all()),
+        store.dispatch(regions.all()),
+        store.dispatch(distributions.all()),
+      ]);
     }
 
     // Necessary to await this for testing
```
Comment (mparke) [inline src/router.js]: what is this line about necessary for testing?

Comment (eatonphil) [inline src/router.js]: It doesn't matter that it happens at any point IRL but in tests we want to make sure everything happened (so we can test things _after_ they happened).

**Pull request merged**